src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -347,17 +430,43 @@ private static Details getDetails(DetailAST ast) {
                     shouldCheckLastRcurly = true;
                     nextToken = getNextToken(ast);
                 }
-                break;
-            case TokenTypes.LITERAL_CATCH:
+            }
+            else if (tokenType == TokenTypes.LITERAL_CATCH) {
                 nextToken = ast.getNextSibling();
                 lcurly = ast.getLastChild();
                 rcurly = lcurly.getLastChild();
                 if (nextToken == null) {
                     shouldCheckLastRcurly = true;
                     nextToken = getNextToken(ast);
                 }
-                break;
-            case TokenTypes.LITERAL_IF:
+
+            }
+            else {
+                shouldCheckLastRcurly = true;
+                nextToken = getNextToken(ast);
+                lcurly = ast.getFirstChild();
+                rcurly = lcurly.getLastChild();
+            }
+            final Details details = new Details();
```

Comment:
In my humble opinion, making class fields might add some unnecessary complexity of having to copy code around. How about instead doing this in the end of each method `return new Details(lcurly, rcurly, nextToken, shouldcheckLastRcurly)`  and creating the corresponding constructor ? 

What do you think ?